fbw-a380x/src/systems/instruments/src/MFD/FMC/fmgc.ts

```diff
@@ -315,6 +315,18 @@ export class FmgcDataService implements Fmgc {
     return (zfw / 1000) * 2204.625;
   }
 
+  /** in kg */
+  public getGrossWeight(): Kilograms {
+    // Value received from FQMS, or falls back to ZFW + FOB
+    const zfw = this.data.zeroFuelWeight.get();
+
+    let fmGW = SimVar.GetSimVarValue('TOTAL WEIGHT', 'kilogram');
```

Comment:
There's no way for the plane to measure it's weight without a ZFW entry + either FoB after engine start, or block entry before engine start. We should not be reading this var.

Reply:
took that logic now